### PR TITLE
Refactor InventoryManager

### PR DESCRIPTION
**Pull request by yuya-yuliya**
- add InventoryManagerV2
- add tests for InventoryManagerV2

### CODE REVIEW

File: src/Foundation/Connect/tests/Managers/Inventory/InventoryManagerV2Tests.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        {
+            var connectServiceProvider = Substitute.For<IConnectServiceProvider>();
+            this.inventoryServiceProvider = Substitute.For<InventoryServiceProvider>();
+
+            connectServiceProvider.GetInventoryServiceProvider().Returns(this.inventoryServiceProvider);
+
+            this.logService = Substitute.For<ILogService<CommonLog>>();
+            this.fixture = new Fixture();
+
+            this.inventoryManager = Substitute.For<InventoryManagerV2>(connectServiceProvider, this.logService);
```
Comment (stanislav-kozorez) [inline src/Foundation/Connect/tests/Managers/Inventory/InventoryManagerV2Tests.cs]: Do tests work correctly with such substitution?

File: src/Foundation/Connect/tests/Managers/Inventory/InventoryManagerV2Tests.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+        private readonly InventoryManagerV2 inventoryManager;
+
+        private readonly IFixture fixture;
+
+        public InventoryManagerV2Tests()
+        {
+            var connectServiceProvider = Substitute.For<IConnectServiceProvider>();
+            this.inventoryServiceProvider = Substitute.For<InventoryServiceProvider>();
+
```
Comment (stanislav-kozorez) [inline src/Foundation/Connect/tests/Managers/Inventory/InventoryManagerV2Tests.cs]: 2 empty lines

File: src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    using DependencyInjection;
+
+    using Providers.Contracts;
+
+    using Sitecore.Commerce.Engine.Connect.Entities;
+    using Sitecore.Commerce.Entities.Inventory;
+    using Sitecore.Commerce.Services.Inventory;
+    using Sitecore.Diagnostics;
+
+    [Service(typeof(IInventoryManagerV2), Lifetime = Lifetime.Transient)]
```
Comment (stanislav-kozorez) [inline src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs]: Let's keep consistency between other managers and use singleton

**Review by stanislav-kozorez — CHANGES_REQUESTED** [review]

File: src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            IEnumerable<CommerceInventoryProduct> inventoryProducts,
+            StockDetailsLevel detailsLevel)
+        {
+            Assert.ArgumentNotNullOrEmpty(shopName, nameof(shopName));
+            Assert.ArgumentNotNull(inventoryProducts, nameof(inventoryProducts));
+            Assert.ArgumentNotNull(detailsLevel, nameof(detailsLevel));
+
+            var request = new GetStockInformationRequest(shopName, inventoryProducts, detailsLevel)
+            {
+                Location = string.Empty
```
Comment (larionovadaria) [inline src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs]: What is that and why it's always empty?

Comment (yuya-yuliya) [reply src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs]: This is a location of store where where stock information requested. We can remove this initialization

Comment (larionovadaria) [reply src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs]: Do we need to pass this one as argument, or get if from context, like it was done in cart manager?

**Review by stanislav-kozorez — APPROVED** [review]

File: src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs (hunk 4 above)
Comment (yuya-yuliya) [reply src/Foundation/Connect/code/Managers/Inventory/InventoryManagerV2.cs]: I think no, because we have no functionality with different stores

**Pull request merged**